homeassistant/components/zwave_js/api.py

```diff
@@ -1188,6 +1191,45 @@ async def get(self, request: web.Request, config_entry_id: str) -> web.Response:
         )
 
 
+class NodeDumpView(HomeAssistantView):
+    """View to dump the state of a node on the Z-Wave JS server."""
+
+    url = r"/api/zwave_js/dump/{config_entry_id}/{node_id:\d+}"
+    name = "api:zwave_js:dump:node"
+
+    async def get(
+        self, request: web.Request, config_entry_id: str, node_id: str
+    ) -> web.Response:
+        """Dump the state of a Z-Wave node."""
+        if not request["hass_user"].is_admin:
+            raise Unauthorized()
+        hass = request.app["hass"]
+
+        if config_entry_id not in hass.data[DOMAIN]:
+            raise web_exceptions.HTTPBadRequest
+
+        client: Client = hass.data[DOMAIN][config_entry_id][DATA_CLIENT]
+        node = client.driver.controller.nodes.get(int(node_id))
+        if not node:
+            raise web_exceptions.HTTPNotFound
+
+        version_info = {
+            "driverVersion": client.version.driver_version,
+            "serverVersion": client.version.server_version,
+            "minSchemaVersion": client.version.min_schema_version,
+            "maxSchemaVersion": client.version.max_schema_version,
+        }
+
+        filename = f"zwave_js_node_{node.node_id}_state.json"
+        return web.Response(
+            body=json.dumps([version_info, node.data], indent=2) + "\n",
+            headers={
+                hdrs.CONTENT_TYPE: "application/json",
+                hdrs.CONTENT_DISPOSITION: f'attachment; filename="{filename}"',
```

Comment:
Any reason we're making this a new endpoint and not just trigger a download via the frontend? That's what we do for the download trace button.

Reply:
Are you suggesting that, instead of creating a JSON file, we pass the data directly to the frontend and let the frontend do the work to convert it into a file?

Reply:
OK found the trace code and I think that is indeed what you are suggesting